### PR TITLE
[CI/Build] Avoid downloading all HF files in `RemoteOpenAIServer`

### PR DESCRIPTION
**Pull request by DarkLight1337**
Fixes https://github.com/vllm-project/vllm/pull/7834#issuecomment-2308396934

### CODE REVIEW

**Review by mgoin — APPROVED** [review]
Nice fix

File: tests/utils.py [hunk 1]
```diff
@@ -60,36 +61,40 @@ class RemoteOpenAIServer:
...
             description="vLLM's remote OpenAI server.")
         parser = make_arg_parser(parser)
-        args = parser.parse_args(cli_args)
+        args = parser.parse_args(["--model", model, *serve_args])
         self.host = str(args.host or 'localhost')
         self.port = int(args.port)
 
+        # download the model before starting the server to avoid timeout
+        engine_args = AsyncEngineArgs.from_cli_args(args)
+        engine_config = engine_args.create_engine_config()
```
Comment (youkaichao) [inline tests/utils.py:85]: can we directly create a load config object? it should be simple in my opinion. just load format auto.

Comment (DarkLight1337) [reply tests/utils.py:85]: Let's fix the tests first before simplifying this.

Comment (DarkLight1337) [reply tests/utils.py:85]: I'm not that familiar with the model loading code so may need some help here.

File: tests/utils.py [hunk 2]
```diff
@@ -60,36 +61,40 @@ class RemoteOpenAIServer:
 
     def __init__(self,
                  model: str,
-                 cli_args: List[str],
+                 serve_args: List[str],
```
Comment (youkaichao) [inline tests/utils.py:64]: why rename the arg name btw?

Comment (DarkLight1337) [reply tests/utils.py:64]: I meant to associate them with `vllm serve` specifically, not just any CLI. Perhaps it could be clearer.

**Review by youkaichao — APPROVED** [review]
thanks for the fix!

**Pull request merged**